### PR TITLE
remove modifing config by reference in multistore

### PR DESCRIPTION
**Pull request by gibkigonzo**
### Short description and why it's useful

it fixes that config is modifying unexpectedly which causes wrong store Id in some cases

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fixed resolving store code on SSR - @andrzejewsky (#3576)
 - Clear user data if error occurs while login - @gibkigonzo (#3588)
 - Fix loading bestsellers on 404 error page - @andrzejewsky (#3540)
+- Remove modifing config by reference in multistore - @gibkigonzo
```
Comment (patzick) [inline CHANGELOG.md]: please add PR number if there is no issue for this
```suggestion
- Remove modifying config by reference in multistore - @gibkigonzo (#3617)
```

File: core/lib/multistore.ts [hunk 2]
```diff
@@ -41,12 +41,12 @@ export function currentStoreView (): StoreView {
...
   let storeView: StoreView = { // current, default store
-    tax: config.tax,
+    tax: Object.assign({}, config.tax),
     i18n: Object.assign({}, config.i18n),
-    elasticsearch: config.elasticsearch,
+    elasticsearch: Object.assign({}, config.elasticsearch),
     storeCode: null,
     storeId: config.defaultStoreCode && config.defaultStoreCode !== '' ? config.storeViews[config.defaultStoreCode].storeId : 1,
-    seo: config.seo || {}
+    seo: Object.assign({}, config.seo || {})
```
Comment (patzick) [inline core/lib/multistore.ts]:
```suggestion
    seo: Object.assign({}, config.seo)
```

File: CHANGELOG.md (hunk 1 above)
Comment (gibkigonzo) [reply CHANGELOG.md]: done

File: core/lib/multistore.ts (hunk 2 above)
Comment (gibkigonzo) [reply core/lib/multistore.ts]: done, thanks :)

**Pull request merged**